src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -169,7 +169,8 @@ public CassandraOnHeapGraph(IndexContext context, boolean forSearching, Memtable
                                         indexConfig.getConstructionBeamWidth(),
                                         indexConfig.getNeighborhoodOverflow(1.0f), // no overflow means add will be a bit slower but flush will be faster
                                         indexConfig.getAlpha(dimension > 3 ? 1.2f : 2.0f),
-                                        indexConfig.isHierarchyEnabled() && jvectorVersion >= 4);
+                                        indexConfig.isHierarchyEnabled() && jvectorVersion >= 4,
+                                        memtable != null); // Skip graph refinement on memtable to speed up flush
```

Comment:
@marianotepper - should we branch here based on whether it is an in memory build? Or do we want this to be configurable at the index level? `memtable` is null iff we are building the graph during compaction.